crates/node/src/types/file.rs

```diff
@@ -8,54 +8,78 @@ use std::path::PathBuf;
 
 pub async fn open_task_file(
     data_dir: &PathBuf,
-    task_id: &str,
     path: &str,
 ) -> Result<tokio::io::BufReader<tokio::fs::File>> {
     let mut path = Path::new(path);
     if path.is_absolute() {
         path = path.strip_prefix("/")?;
     }
-    let path = PathBuf::new().join(data_dir).join(task_id).join(path);
+    let path = PathBuf::new().join(data_dir).join(path);
```

Comment:
Given I read the data flow correctly, here the path that VM returns is something that node is responsible for. It must not be this way. The VM must not have any idea of how the data is managed on the node side.